src/wp-includes/revision.php

```diff
@@ -516,6 +516,53 @@ function wp_get_post_revisions( $post_id = 0, $args = null ) {
 	return $revisions;
 }
 
+/**
+ * Returns last revisions of specified post and count of total revisions.
+ *
+ * @since 6.1.0
+ *
+ * @param int|WP_Post $post Optional. Post ID or post object. Default is the global $post.
+ * @return array {
+ *     Returns associative array with last revision and total count.
+ *
+ *     @type int $revision The last revision post id or 0 if non existing.
+ *     @type int $count The total count of revisions for $post_id.
```

Comment:
Thinking about the return signature.

The same think is returned if `$post` does not exist and if `$post` exists but there are no revisions. Should these differ?

Reply:
Agreed. FWIW I think it'd be better to optionally return a `WP_Error` instance here for both failure cases.

Reply:
```
if ( ! $post || ! wp_revisions_enabled( $post ) ) {
		return new WP_Error( 'revision_error', __( 'Invalid post or revisions not enabled.' ) );
	}
```

Reply:
Yes, IMO those should be two separate error codes.